### PR TITLE
Added '--prefill' command line option to WSL.

### PR DESCRIPTION
**Pull request by cnihelton**
Hi @jpnurmi !

I am working on prefilling the WSL OOBE with user and locale information acquired from Windows through the launcher, as you may know. Since in WSL Subiquity is started by the OOBE I had to touch this code base. Could you take some time to code review it?

The proposed changes do the following:
- Check if the file passed by CLI exists before passing to the server (not sure if this kind of check belongs here, though).
- When server starts, log.info also it's CLI options.

### CODE REVIEW

File: packages/subiquity_client/lib/subiquity_server.dart [hunk 1]
```diff
@@ -94,7 +94,9 @@ abstract class SubiquityServer {
       stderr.addStream(process.stderr);
       return process;
     });
-    log.info('Starting server (PID: ${_serverProcess!.pid})');
+    log.info(
+      'Starting server (PID: ${_serverProcess!.pid}) with args: $subiquityCmd',
```
Comment (jpnurmi) [inline packages/subiquity_client/lib/subiquity_server.dart:98]: this is nice. :+1: it would be nice to have it in a separate commit, though

File: packages/ubuntu_wsl_setup/lib/main.dart [hunk 2]
```diff
@@ -7,11 +9,26 @@ import 'app.dart';
...
   })!;
   var variant = Variant.WSL_SETUP;
+  List<String>? serverArgs;
   if (options['reconfigure'] == true) {
     variant = Variant.WSL_CONFIGURATION;
   }
+  if (options['prefill'] != null) {
+    final prefillFile = options['prefill'];
+    final fileExists = File(prefillFile).existsSync();
+    if (fileExists) serverArgs = ['--prefill', options['prefill']];
```
Comment (jpnurmi) [inline packages/ubuntu_wsl_setup/lib/main.dart:30]: should we print an error if the file doesn't exist, to avoid silently ignoring the option?

how does the server handle it if the file doesn't exist? I'm wondering if simply passing the option through and letting the server throw an error would result in a clear enough error message?

File: packages/subiquity_client/lib/subiquity_server.dart (hunk 1 above)
Comment (CarlosNihelton) [reply packages/subiquity_client/lib/subiquity_server.dart:98]: I'll do some git tricks to split that commit in two.

File: packages/ubuntu_wsl_setup/lib/main.dart [hunk 3]
```diff
@@ -7,11 +7,24 @@ import 'app.dart';
 void main(List<String> args) {
   final options = parseCommandLine(args, onPopulateOptions: (parser) {
     parser.addFlag('reconfigure');
+    parser.addOption(
+      'prefill',
+      valueHelp: 'prefill',
+      help: '''
```
Comment (jpnurmi) [inline packages/ubuntu_wsl_setup/lib/main.dart:13]: I'd suggest also `hide: true` since it's not for users (same as `initial-route`) which is for testing.

**Review by jpnurmi — APPROVED** [review]
LGTM.

Feel free to ignore the suggestion if you prefer to keep the option visible in `--help`.

File: packages/ubuntu_wsl_setup/lib/main.dart (hunk 3 above)
Comment (CarlosNihelton) [reply packages/ubuntu_wsl_setup/lib/main.dart:13]: Keeping it visible might be useful for development/testing. Let's keep it as is.

Thank you again, JP!
